src/bidiMapper/scripts/eval.es

```diff
@@ -16,117 +16,147 @@
  */
 
 (() => {
-  class Serializer {
+  class SerializationMapper {
     _idToObject = new Map();
     _objectToId = new WeakMap();
+  }
 
-    constructor() {}
-
-    serialize(value) {
-      if (value === undefined) {
-        return { type: 'undefined' };
-      } else if (value === null) {
-        return { type: 'null' };
-      } else if (typeof value === 'string') {
-        return { type: 'string', value };
-      } else if (typeof value === 'number') {
-        let serialized;
-        if (Number.isNaN(value)) {
-          serialized = 'NaN';
-        } else if (Object.is(value, -0)) {
-          serialized = '-0';
-        } else if (value === -Infinity) {
-          serialized = '-Infinity';
-        } else if (value === +Infinity) {
-          serialized = '+Infinity';
-        } else {
-          serialized = value;
-        }
-        return { type: 'number', value: serialized };
-      } else if (typeof value === 'boolean') {
-        return { type: 'boolean', value };
-      } else if (value instanceof Object) {
-        // TODO: Recursive serialize.
-        let objectId = this._objectToId.get(value);
-        if (!objectId) {
-          objectId = uuid();
-          this._objectToId.set(value, objectId);
-          this._idToObject.set(objectId, new WeakRef(value));
-        }
-        return { type: 'object', objectId };
+  function getSerializationMapper() {
+    // No `window` can be in case of unit tests.
```

Comment:
Can we use `globalThis` instead of `window`?

Reply:
done